src/Microsoft.AspNet.Security.OAuth/Notifications/OAuthBearerAuthenticationNotifications.cs

```diff
@@ -3,64 +3,61 @@
 
 using System;
 using System.Threading.Tasks;
+using Microsoft.AspNet.Security.Notifications;
 
+/// <summary>
+/// Specifies events which the <see cref="OAuthBearerAuthenticationMiddleware"></see> invokes to enable developer control over the authentication process. />
+/// </summary>
 namespace Microsoft.AspNet.Security.OAuth
 {
     /// <summary>
     /// OAuth bearer token middleware provider
     /// </summary>
-    public class OAuthBearerAuthenticationNotifications : IOAuthBearerAuthenticationNotifications
+    public class OAuthBearerAuthenticationNotifications
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="OAuthBearerAuthenticationProvider"/> class
         /// </summary>
         public OAuthBearerAuthenticationNotifications()
         {
-            OnRequestToken = context => Task.FromResult<object>(null);
-            OnValidateIdentity = context => Task.FromResult<object>(null);
-            OnApplyChallenge = context =>
-            {
-                context.HttpContext.Response.Headers.AppendValues("WWW-Authenticate", context.Challenge);
-                return Task.FromResult(0);
-            };
-        }
-
-        /// <summary>
-        /// Handles processing OAuth bearer token.
-        /// </summary>
-        public Func<OAuthRequestTokenContext, Task> OnRequestToken { get; set; }
+			AuthenticationFailed = notification => Task.FromResult(0);
```

Comment:
You can easily replace with constructor with the new inline properties setting stuff offered by Roslyn.